### PR TITLE
Logout and redirect to login when jupyterhub token expires

### PR DESCRIPTION
**Pull request by vfried**
This addresses the bug #370 (It solves the problem following the instructions that @ableuler describes in the bug) This is a way of "fake-replicating" the bug so we can't be 100% sure that it will work, but it should.

### CODE REVIEW

File: src/api-client/index.js [hunk 1]
```diff
@@ -58,7 +57,6 @@ class APIClient {
     addUserMethods(this);
     addKuMethods(this);
     addInstanceMethods(this);
-    addNotebookServersMethods(this);
```
Comment (ableuler) [inline src/api-client/index.js]: I don't understand why you are removing this. Won't this break the notebook admin tab in the projects page that @lorenzo-cavazzi created recently?

Comment (vfried) [reply src/api-client/index.js]: I know, i made a mistake when merging things... I will correct it tomorrow morning.

Comment (vfried) [reply src/api-client/index.js]: I just corrected this...

**Comment by rokroskar** [comment]
@vfried just to be sure, this is using the response from the notebooks service implemented in https://github.com/SwissDataScienceCenter/renku-notebooks/pull/117 correct?

**Comment by vfried** [comment]
> @vfried just to be sure, this is using the response from the notebooks service implemented in [SwissDataScienceCenter/renku-notebooks#117](https://github.com/SwissDataScienceCenter/renku-notebooks/pull/117) correct?

Yes, it's using #117

File: src/notebooks/Notebooks.container.js [hunk 2]
```diff
@@ -50,25 +51,27 @@ class LaunchNotebookServer extends Component {
...
-    this.componentDidUpdate()
+    if (this.state.doLogOut === false)
+      this.componentDidUpdate()
   }
 
   componentDidUpdate() {
-    if (this.props.core.notebookServerAPI !== this.previousNotebookServerAPI){
-      this.serverStatusSet = false;
-      this.serverOptionsSet = false;
+    if (this.state.doLogOut === false) {
```
Comment (ciyer) [inline src/notebooks/Notebooks.container.js]: I think this logic would be easier to follow if it were:
```
if (this.state.doLogOut === true) return;  // We are about to refresh anyway...
[same code as before]
```

Comment (vfried) [reply src/notebooks/Notebooks.container.js]: You are right...

**Comment by rokroskar** [comment]
If this works, can we approve, merge and make a bugfix release of the UI and the notebooks service?

**Comment by ciyer** [comment]
> If this works, can we approve, merge and make a bugfix release of the UI and the notebooks service?

Yes, good idea.

**Pull request merged**